app/src/main/java/com/keylesspalace/tusky/fragment/TimelineFragment.kt

```diff
@@ -0,0 +1,1273 @@
+                TimelineRequestMode.NETWORK)
+                .observeOn(AndroidSchedulers.mainThread())
+                .autoDispose(from(this))
+                .subscribe(
+                        { statuses: List<Either<Placeholder, Status>> ->
+
+                            initialUpdateFailed = false
+                            // When cached timeline is too old, we would replace it with nothing
+                            if (statuses.isNotEmpty()) {
+                                val mutableStatuses = statuses.toMutableList()
+                                filterStatuses(mutableStatuses)
+                                if (!this.statuses.isEmpty()) {
+                                    // clear old cached statuses
+                                    val iterator = this.statuses.iterator()
+                                    while (iterator.hasNext()) {
+                                        val item = iterator.next()
+                                        if (item.isRight()) {
+                                            val (id1) = item.asRight()
+                                            if (id1.length < topId.length || id1 < topId) {
+                                                iterator.remove()
+                                            }
+                                        } else {
+                                            val (id1) = item.asLeft()
+                                            if (id1.length < topId.length || id1 < topId) {
+                                                iterator.remove()
+                                            }
+                                        }
+                                    }
+                                }
+                                this.statuses.addAll(mutableStatuses)
+                                updateAdapter()
+                            }
+                            bottomLoading = false
+                        },
+                        { t: Throwable? ->
+                            Log.d(TAG, "Failed updating timeline", t)
+                            initialUpdateFailed = true
+                            // Indicate that we are not loading anymore
+                            binding.progressBar.hide()
+                            binding.swipeRefreshLayout.isRefreshing = false
+                        })
+    }
+
+    private fun setupTimelinePreferences() {
+        alwaysShowSensitiveMedia = accountManager.activeAccount!!.alwaysShowSensitiveMedia
+        alwaysOpenSpoiler = accountManager.activeAccount!!.alwaysOpenSpoiler
+        if (kind == Kind.HOME) {
+            val preferences = PreferenceManager.getDefaultSharedPreferences(context)
+            filterRemoveReplies = !preferences.getBoolean("tabFilterHomeReplies", true)
+            filterRemoveReblogs = !preferences.getBoolean("tabFilterHomeBoosts", true)
+        }
+        reloadFilters(false)
```

Comment:
this complains, should change in SFragment to

```java
    @VisibleForTesting(otherwise = VisibleForTesting.PROTECTED)
    public void reloadFilters(boolean forceRefresh) {
      //...
    }
```